docs/framework/guides/deep-dive/localization.md

```diff
@@ -0,0 +1,240 @@
+} );
+// ...
+```
+
+<info-box warning>
+	Note that this code sample lacks a few parts. See {@link framework/guides/creating-simple-plugin how to create a complete plugin} to have a better understanding about creating CKEditor 5 plugins.
+</info-box>
+
+### Example: Localizing `aria` attributes
+
+For better accessibility support, do not forget about making `aria` attributes localizable:
+
+```js
+editingView.change( writer => {
+	const editingView = this._editingView;
+	const t = this.t;
+
+	editingView.change( writer => {
+		const viewRoot = editingView.document.getRoot( this.name );
+
+		writer.setAttribute( 'aria-label', t( 'Rich Text Editor, %0', [ this.name ] ), viewRoot );
+	} );
+} );
+```
+
+### Example: Localizing pending actions
+
+{@link module:core/pendingactions~PendingActions Pending actions} are used to inform the user that an action is in progress and they will lose data if they exit the editor at the given moment. Here is how you can localize them:
+
+```js
+class FileRepository {
+	// ...
+	updatePendingAction() {
+		// ...
+		const pendingActions = this.editor.plugins.get( PendingActions );
+
+		// ...
+		const t = this.editor.t;
+		const getMessage = value => t( 'Upload in progress (%0%).', value ); // Upload in progress (12%).
+
+		this._pendingAction = pendingActions.add( getMessage( this.uploadedPercent ) );
+		this._pendingAction.bind( 'message' ).to( this, 'uploadedPercent', getMessage );
+	}
+}
+```
+
+## Adding translations and localizing the editor UI
+
+First of all, if you have found a missing or incorrect translation in any of CKEditor 5 features, {@link framework/guides/contributing/contributing#translating see how you can contribute to the project}! CKEditor 5 is an Open Source project used by people from all around the world, so your help will be appreciated by others.
+
+Adding translations to the editor can be done in three ways to satisfy various needs.
+
+- By [adding translations via the translation-service's `add()` function](#using-the-add-function).
+	This can be done before initiating the CKEditor 5 editor instance but it requires importing the CKEditor 5 utility function.
+- By [extending the global `window.CKEDITOR_TRANSLATIONS` object](#using-the-windowckeditor_translations-object).
+	This can be done before initiating the CKEditor 5 editor instance.
+- By [creating `.po` files with translations](#creating-po-files) in the `lang/translations/` directory of the published package like other CKEditor 5 packages do.
+	This option will be useful for third-party plugin creators as it allows bundling translations only for needed languages during the webpack compilation.
+
+### Using the `add()` function
+
+The first option for adding translations is via the {@link module:utils/translation-service.add the translation-service's `add()` helper}. This utility adds translations to the `window.CKEDITOR_TRANSLATIONS` object by extending it. Since it needs to be imported, it works only before building the editor.
```

Comment:
```suggestion
The first option for adding translations is via {@link module:utils/translation-service.add the translation-service's `add()` helper}. This utility adds translations to the `window.CKEDITOR_TRANSLATIONS` object by extending it. Since it needs to be imported, it works only before building the editor.
```